src/BloomExe/Book/Book.cs

```diff
@@ -2397,5 +2397,20 @@ public static string MakeVersionCode(string fileContent, string filePath = null)
 			sha.TransformFinalBlock(new byte[0], 0, 0);
 			return Convert.ToBase64String(sha.Hash);
 		}
+
+		public void MakeThumbnailFromCoverPicture()
+		{
+			var coverImgElt = _storage.Dom.SafeSelectNodes("//div[@data-book='coverImage' and @style]").Cast<XmlElement>().FirstOrDefault();
```

Comment:
Better to use HtmlDom.GetImageElementUrl(), which encapsulates this logic and makes it easier in the future to change how we do images, because we only have to change one place.